lib/connection_controller.go

```diff
@@ -5,12 +5,23 @@ import (
 	"github.com/btcsuite/btcd/addrmgr"
 	"github.com/btcsuite/btcd/wire"
 	"github.com/deso-protocol/core/bls"
+	"github.com/deso-protocol/core/collections"
 	"github.com/golang/glog"
 	"github.com/pkg/errors"
 	"net"
 	"strconv"
+	"sync"
+	"time"
 )
 
+type GetActiveValidatorsFunc func() *collections.ConcurrentMap[bls.SerializedPublicKey, *ValidatorEntry]
+
+var GetActiveValidatorImpl GetActiveValidatorsFunc = BasicGetActiveValidators
```

Comment:
Is this going to be used outside of test files? If it's only going to be used in test files, let's move it to one of the test files instead of sticking it in the connection controller file. I know we've used this pattern in the past for the encoder migration tests, but it's not my favorite as there's extra overhead to figure out what's actually happening.

Reply:
Good call! We figured out a better approach based on pushing the latest validator set to the NetworkManager (ConnectionController) directly from consensus. This approach resembles how Mempool receives the latest UtxoView, if you remember how we solved it there.

Reply:
It's in #945 btw.